File: sdk/src/types/pagination.ts
/** Pagination metadata attached to every `PaginatedResult`. */
export interface PaginationInfo {
  total: number;
  page: number;
  limit: number;
  pages: number;
}

/**
 * Envelope returned by every paginated `list()`/`discover()` method
 * (Agents/Skills/Knowledge/MCP/Threads/Files — not Providers, which has no
 * discovery concept and stays a bare array).
 */
export interface PaginatedResult<T> {
  items: T[];
  pagination: PaginationInfo;
}

File: sdk-python/src/personaai/types/pagination.py
"""Mirrors ``sdk/src/types/pagination.ts``."""

from __future__ import annotations

from typing import Generic, TypedDict, TypeVar

T = TypeVar("T")


class PaginationInfo(TypedDict):
    total: int
    page: int
    limit: int
    pages: int


class PaginatedResult(TypedDict, Generic[T]):
    """Envelope returned by every paginated ``list()``/``discover()`` method
    (Agents/Skills/Knowledge/MCP/Threads/Files — not Providers, which has no
    discovery concept and stays a bare list)."""

    items: list[T]
    pagination: PaginationInfo
